wiring/inc/spark_wiring_spi.h

```diff
@@ -217,24 +224,34 @@ class SPIClass {
 
   bool trylock()
   {
-#if PLATFORM_THREADING
-    return mutex_.trylock();
+#if HAL_PLATFORM_SPI_HAL_THREAD_SAFETY
+    return true;
+#elif PLATFORM_THREADING
+    return _mutex.trylock();
 #else
+    //TODO: Implement by extending HAL_SPI_Acquire with immediate timeout
     return true;
 #endif
   }
 
-  void lock()
+  int32_t lock()
   {
-#if PLATFORM_THREADING
-    mutex_.lock();
+#if HAL_PLATFORM_SPI_HAL_THREAD_SAFETY
+    return HAL_SPI_Acquire(_spi, nullptr);
+#elif PLATFORM_THREADING
+    _mutex.lock();
+    return 0;
+#else
+    return 0;
 #endif
   }
 
   void unlock()
```

Comment:
What would an `unlock()` error look like, and how could a user respond to an `unlock()` error?

Reply:
I agree, it doesn't make much sense, since potentially the only errors that could be returned are when the lock was not held, or when it was unlocked by a wrong thread, so maybe let's leave it as-is, however for consistency I think it's either
```
int lock();
bool tryLock();
int unlock();
```

or 
```
void lock();
bool tryLock();
void unlock();
```

Let's leave as-is.